### PR TITLE
feat: handle `accountNameSuggestion` and `displayConfirmation` options in `#handleAccountCreated` 

### PR DESCRIPTION
**Pull request by k-g-j**
## Description

- Bumps the Keyring API to latest version (6.2.0) that has the optional fields of `accountNameSuggestion` and `displayConfirmation` in the [AccountCreatedEventStruct](https://github.com/MetaMask/keyring-api/blob/main/src/internal/events.ts#L9-L32)
- `#handleAccountCreated` now parses these new options from the message params and passes them along with other properties to the add account callback 
- Tests are added for the new scenarios of creating an account with a suggested name or display confirmation option

Dependent on [PR 301](https://github.com/MetaMask/eth-snap-keyring/pull/301)

### CODE REVIEW

File: src/SnapKeyring.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ import type {
   KeyringResponse,
   KeyringExecutionContext,
   BtcMethod,
+  ExactOptionalTag,
```
Comment (danroc) [inline src/SnapKeyring.ts:28]: This shouldn't be needed, it's probably a bug in the Keyring API :(

File: src/SnapKeyring.ts [hunk 2]
```diff
@@ -83,6 +84,8 @@ export type SnapKeyringCallbacks = {
     address: string,
     snapId: SnapId,
     handleUserInput: (accepted: boolean) => Promise<void>,
+    accountNameSuggestion?: string | ExactOptionalTag,
```
Comment (danroc) [inline src/SnapKeyring.ts:87]: Same here, I pushed a fix to the Keyring API, will make a new release.

File: src/SnapKeyring.test.ts
Comment (danroc) [inline src/SnapKeyring.test.ts:1]: I will wait until the test-reorg PR gets approved to review this one 😄

File: src/SnapKeyring.ts (hunk 1 above)
Comment (danroc) [reply src/SnapKeyring.ts:28]: It should be ok to bump the `keyring-api` to 6.2.1, it should solve the issue.

File: src/SnapKeyring.test.ts
Comment (k-g-j) [reply src/SnapKeyring.test.ts:1]: Reorg PR approved! https://github.com/MetaMask/eth-snap-keyring/pull/301

File: src/SnapKeyring.test.ts [hunk 3]
```diff
@@ -643,6 +694,38 @@ describe('SnapKeyring', () => {
       );
       spy.mockRestore();
     });
+    const isNotAllowedOrigin = async (
```
Comment (ccharly) [inline src/SnapKeyring.test.ts:697]: Any reason why this one has been moved here now?

Comment (k-g-j) [reply src/SnapKeyring.test.ts]: Got messed up in the merged! Fixed here [2373c78](https://github.com/MetaMask/eth-snap-keyring/pull/300/commits/2373c783c0184c8ac5d40a9ecf63645cea63af99)

File: src/SnapKeyring.test.ts [hunk 4]
```diff
@@ -203,6 +272,19 @@ describe('SnapKeyring', () => {
         );
       });
 
+      it('cannot update an account owned by another Snap', async () => {
+        await expect(
+          keyring.handleKeyringSnapMessage('a-different-snap-id' as SnapId, {
+            method: KeyringEvent.AccountCreated,
```
[the remark this answers is not in the document]
Comment (k-g-j) [reply src/SnapKeyring.test.ts]: Good call! I updated the naming here and moved the test to the proper location 
[c39badc](https://github.com/MetaMask/eth-snap-keyring/pull/300/commits/c39badcc7ede47304e029cf54e7958da45168a9a)

**Review by ccharly — APPROVED** [review]

**Pull request merged**